tests/bluetooth/gatt/src/main.c

```diff
@@ -142,6 +142,157 @@ ZTEST(test_gatt, test_gatt_unregister)
 		     "Test service unregister failed");
 }
 
+ZTEST(test_gatt, test_gatt_reregister)
+{
+	struct bt_gatt_attr local_test_attrs[] = {
+		/* Vendor Primary Service Declaration */
+		BT_GATT_PRIMARY_SERVICE(&test_uuid),
+
+		BT_GATT_CHARACTERISTIC(&test_chrc_uuid.uuid, BT_GATT_CHRC_READ | BT_GATT_CHRC_WRITE,
+				       BT_GATT_PERM_READ_AUTHEN | BT_GATT_PERM_WRITE_AUTHEN,
+				       read_test, write_test, test_value),
+	};
+
+	struct bt_gatt_attr local_test1_attrs[] = {
+		/* Vendor Primary Service Declaration */
+		BT_GATT_PRIMARY_SERVICE(&test1_uuid),
+
+		BT_GATT_CHARACTERISTIC(&test1_nfy_uuid.uuid, BT_GATT_CHRC_NOTIFY, BT_GATT_PERM_NONE,
+				       NULL, NULL, &nfy_enabled),
+		BT_GATT_CCC(test1_ccc_cfg_changed, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
+	};
+	struct bt_gatt_service local_test_svc = BT_GATT_SERVICE(local_test_attrs);
+	struct bt_gatt_service local_test1_svc = BT_GATT_SERVICE(local_test1_attrs);
+
+	for (int i = 0; i < 10; i++) {
+
+		/* Attempt to register service A */
+		zassert_false(bt_gatt_service_register(&local_test_svc),
+			      "Test service A registration failed");
+
+		/* Attempt to unregister service A */
+		zassert_false(bt_gatt_service_unregister(&local_test_svc),
```

Comment:
The registration function may at some point be rewritten in such a way that the test still passes, but the handles are not reset. Therefore it is better to check that all handles are zero-ed after unregistering the service. This will guarantee that we get exactly what we want.

Reply:
I removed this check because I thought it was a test of the unregister function, which should probably not be done inside this test. Should I still add the check here or should this be added to the test of the unregister function instead?

Reply:
Yes, I think it makes sense to check handles here.